### PR TITLE
Era change can be forced on and forced off

### PR DESCRIPTION
**Pull request by gavofyork**
Before it was only possible to force an era. Now we can force eras not to happen, useful for the PoA period of Kusama.

### CODE REVIEW

File: srml/staking/src/lib.rs [hunk 1]
```diff
@@ -555,6 +555,22 @@ pub trait Trait: system::Trait {
...
+/// Mode of era-forcing.
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(Debug, Serialize, Deserialize))]
+pub enum Forcing {
+	/// Not forcing anything - just let whatever happen.
+	NotForcing,
+	/// Force a new era, then reset to `NotForcing` as soon as it is done.
+	ForceNew,
+	/// Avoid a new era indefinitely.
+	ForceNone,
```
Comment (rphmeier) [inline srml/staking/src/lib.rs]: more idiomatic would be to have `Forcing::New` and `Forcing::Avoid`, with storage holding an `Option<Forcing>`

Comment (rphmeier) [reply srml/staking/src/lib.rs]: (typical Rust style would call this `Force` as opposed to `Forcing`)

Comment (gavofyork) [reply srml/staking/src/lib.rs]: yeah, didn't manage to make that work with the module JSON config stuff. let me try again though - might have been some other issue.

Comment (gavofyork) [reply srml/staking/src/lib.rs]: yeah, genesis configs require `Option` items to be `Some` value.

Comment (gavofyork) [reply srml/staking/src/lib.rs]: See branch `gav-disable-eras-alt`

Comment (rphmeier) [reply srml/staking/src/lib.rs]: OK, that's a bit unfortunate. but no reason to block this PR on fixing the macro

Comment (rphmeier) [reply srml/staking/src/lib.rs]: Regardless, the repetition of `Force` in the variant names is redundant and it would be a bit cleaner without
